### PR TITLE
nixos/asf: fix home directory permissions, for real

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/games/asf.nix [hunk 1]
```diff
@@ -158,6 +158,7 @@ in
 
     users = {
       users.asf = {
+        createHome = true;
```
Comment (legendofmiracles) [inline nixos/modules/services/games/asf.nix:161]: There was a lot of discussion about this in the initial PR and it was determined to be better to use StateDirectory

See here: https://github.com/NixOS/nixpkgs/pull/148541#discussion_r762430559

Comment (SuperSandro2000) [reply nixos/modules/services/games/asf.nix]: Oh, yeah that makes a lot of sense. I also reverted one commit from yesterday.

File: nixos/modules/services/games/asf.nix [hunk 2]
```diff
@@ -174,13 +173,17 @@ in
         wantedBy = [ "multi-user.target" ];
 
         serviceConfig = mkMerge [
-          (mkIf (cfg.dataDir == "/var/lib/asf") { StateDirectory = "asf"; })
+          (mkIf (cfg.dataDir == "/var/lib/asf") {
+            StateDirectory = "asf";
+            StateDirectoryMode = "700";
```
Comment (SuperSandro2000) [inline nixos/modules/services/games/asf.nix:178]: I am not sure if this should be 700 or 750 but I am really sure that the last bit must be a 0.

**Comment by legendofmiracles** [comment]
I only just finished compiling your first PR.. and the service failed with `touch config/bot: permission denied` (paraphrased).
Is that what reverting the commit fixes?
I don't think this module has many users either way, pushing something breaking (for a much better module in the long term) shouldn't be the end of the world

File: nixos/modules/services/games/asf.nix (hunk 2 above)
Comment (mweinelt) [reply nixos/modules/services/games/asf.nix:179]: the last octet ;)

**Comment by SuperSandro2000** [comment]
> Is that what reverting the commit fixes?

No, the reverted commit was actually a noop.

The error you got was probably caused by not setting the default correct and not actually testing the default locally.

**Review by SuperSandro2000 — COMMENTED** [review]
I assume your testing was successful and we can merge this? Your change is LGTM to me.

**Comment by SuperSandro2000** [comment]
btw is the log tab for you working? For me it just shows blank.

**Comment by legendofmiracles** [comment]
I've tried it again, deleted all my asf stuff and restarted it with what is running on 22.05. Then when I updated to this branch it fails with `cp: cannot create regular file 'config/ASF.json': Read-only file system`. Only after I remove /var/lib/asf does it work.

The log tab still works for me, with and before this patch.

Also, I've forced pushed my commit again because the copying of the file wasn't optional resulting in an error if ipcPasswordFile was not set

**Comment by SuperSandro2000** [comment]
> The log tab still works for me, with and before this patch.

I just checked my browser console and there is an error. Probably unrelated to this.

**Pull request merged**